lib/lighthouse/benefits_documents/configuration.rb

```diff
@@ -61,16 +61,34 @@ def post(body, document_data, lighthouse_client_id = nil, lighthouse_rsa_key_pat
       }",
                   'Content-Type' => 'multipart/form-data' }
 
-      connection.post(DOCUMENTS_PATH, body, headers) do |req|
-        req.params['systemName'] = SYSTEM_NAME
-        req.params['docType'] = document_data.document_type
-        req.params['claimId'] = document_data.claim_id
-        req.params['fileNumber'] = document_data.file_number
-        # In theory one document can correspond to multiple tracked items
-        # To do that, add multiple query parameters
-        req.params['trackedItemIds'] = document_data.tracked_item_id
-        req.params['fileName'] = document_data.file_name
-      end
+      body = generate_upload_body(document_data, file_body)
+      connection.post(DOCUMENTS_PATH, body, headers)
+    end
+
+    def generate_upload_body(document_data, file_body)
+      payload = {}
+      data = {
+        data: {
+          systemName: SYSTEM_NAME,
+          docType: document_data[:document_type],
+          claimId: document_data[:claim_id],
+          fileNumber: document_data[:file_number],
+          fileName: document_data[:file_name],
+          # In theory one document can correspond to multiple tracked items
+          # To do that, add multiple query parameters
+          trackedItemIds: document_data[:tracked_item_id]
+        }
+      }
+
+      payload[:parameters] = data
+      fn = Tempfile.new('params')
+      File.write(fn, data.to_json)
+      payload[:parameters] = Faraday::UploadIO.new(fn, 'application/json')
+
+      file = Tempfile.new(document_data[:file_name])
+      File.write(file, file_body)
+      payload[:file] = Faraday::UploadIO.new(file, 'application/pdf')
```

Comment:
I'm still shocked it requires this amount of workaround to get Faraday and this endpoint to play nicely

Reply:
Yeah, this is definitely non-obvious. IDK why you have to write the params to a tempfile. That seems pretty unnintuitive. Maybe there's a simpler way, but hey if it works...